napi_js_data.md

```diff
@@ -0,0 +1,760 @@
+- `[out] data`: The underlying data buffer of the ArrayBuffer
+- `[out] byte_length`: Length in bytes of the underlying data buffer
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API is used to retrieve the underlying data buffer of an ArrayBuffer and it's length.
+WARNING: This API is dangerous. The lifetime of the underlying data buffer is managed by the 
+ArrayBuffer even after it's returned to you. The one possible safe to use this API is in conjunction 
+with napi_create_reference, which can be used to guarantee that you control the lifetime of
+the ArrayBuffer.
+
+<TODO: What happens to this buffer in case of a compacting GC?>
+
+### *napi_get_boolean*
+
+#### Signature
+```
+napi_status napi_get_boolean(napi_env env, bool value, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: The value of the boolean to retrieve
+- `[out] result`: `napi_value` representing JavaScript Boolean singleton to retrieve
+
+#### Description
+This API is used to return the JavaScript singleton object that is used to represent the given boolean value
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_buffer_info*
+
+#### Signature
+```
+napi_status napi_get_buffer_info(napi_env env,
+                                 napi_value value,
+                                 void** data,
+                                 size_t* length)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: napi_value representing the node::Buffer being queried
+- `[out] data`: The underlying data buffer of the node::Buffer
+- `[out] length`: Length in bytes of the underlying data buffer
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API is used to retrieve the underlying data buffer of an node::Buffer and it's length.
+
+### *napi_get_global*
+
+#### Signature
+```
+napi_status napi_get_global(napi_env env, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[out] result`: `napi_value` representing JavaScript Global Object
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_null*
+
+#### Signature
+```
+napi_status napi_get_null(napi_env env, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[out] result`: `napi_value` representing JavaScript Null Object
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_prototype*
+
+#### Signature
+```
+napi_status napi_get_prototype(napi_env env, napi_value object, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  object`: `napi_value` representing JavaScript Object whose prototype to return
+- `[out] result`: `napi_value` representing prototype of the given object
```

Comment:
Add a Description section that clarifies that this returns the equivalent of `Object.getPrototypeOf()`, which is _not_ the same as a function's "prototype" property.